pybamm_tea/tea.py

```diff
@@ -394,223 +425,191 @@ def print_capacities_and_potentials(self):
-        # Create the DataFrame from the list of dictionaries
+
+        names = []
+        units = []
+        values = []
+        for parameter in parameters:
+            if "[" in parameter:
+                name, unit = parameter.split("[")
+                names.append(name.strip())
+                units.append(unit.rstrip(']'))
+            else:
+                names.append(parameter.strip())
+                units.append("-")
+            values.append(stack_ed.get(parameter))
+
+        data = {
+            "Parameter": names,
+            "Unit": units,
+            "Value": values,
+            }
+        
         df = pd.DataFrame(data)
         return df
-
-    def calculate_stack_energy_densities(self):
+        
+    
+    def calculate_stack_energy(self):
         """
         Calculate ideal volumetric and gravimetric energy densities on stack level.
         """
         stack_ed = {}  # stack energy densities dict
-        pava = None
-        pava = self.parameter_values  # parameter values
+        pava = dict(self.parameter_values)  # parameter values
 
         # stoichiometries - calculation based on input stoichiometries or cell
         # potential limits
         if (
-            pava.get("Negative electrode stoichiometry at 0%") is not None
-            and pava.get("Negative electrode stoichiometry at 100%") is not None
-            and pava.get("Positive electrode stoichiometry at 100%") is not None
-            and pava.get("Positive electrode stoichiometry at 0%") is not None
+            pava.get("Negative electrode stoichiometry at 0% SoC") is not None
+            and pava.get("Negative electrode stoichiometry at 100% SoC") is not None
+            and pava.get("Positive electrode stoichiometry at 100% SoC") is not None
+            and pava.get("Positive electrode stoichiometry at 0% SoC") is not None
         ) or (
-            pava.get("Negative electrode stoichiometry at 0%") is None
-            and pava.get("Negative electrode stoichiometry at 100%") is None
-            and pava.get("Positive electrode stoichiometry at 100%") is not None
-            and pava.get("Positive electrode stoichiometry at 0%") is not None
+            pava.get("Negative electrode stoichiometry at 0% SoC") is None
+            and pava.get("Negative electrode stoichiometry at 100% SoC") is None
+            and pava.get("Positive electrode stoichiometry at 100% SoC") is not None
+            and pava.get("Positive electrode stoichiometry at 0% SoC") is not None
         ):
             y100 = pava.get("Positive electrode stoichiometry at 100% SoC")
             y0 = pava.get("Positive electrode stoichiometry at 0% SoC")
-            # x0, x100 = self.get_stoichiometries(pava, y0, y100)
-            # update cut-off voltages if voltage curve(s) are provided
+            # calculate lithium in positive electrode at SOC = 0
+            n_pe_0 = y0 * (pava.get("Maximum concentration in positive electrode [mol.m-3]")
```

Comment:
Would be cleaner to have already stored the electrode capacities, then you can just get those from the dict instead of repeating the calculation all the time